lib/cache/entry.js

```diff
@@ -274,6 +274,10 @@ class CacheEntry {
       const cacheWritePromise = new Promise((resolve, reject) => {
         cacheWriteResolve = resolve
         cacheWriteReject = reject
+      }).catch((err) => {
+        if (body) {
```

Comment:
What if there is no body? Do we still need to emit an error event so the stream is cleaned up?

Reply:
@hashtagchris pointed out that there will always be a body in this so I removed the if statement guard.